hamilflow/models/utils/typing.py

```diff
@@ -0,0 +1,5 @@
+from typing import Sequence, TypeVar
+
+from numpy.typing import ArrayLike
+
+TypeTime = TypeVar("TypeTime", Sequence[float], Sequence[int], ArrayLike)
```

Comment:
```suggestion
TypeTime = TypeVar("TypeTime", Sequence[float], "ArrayLike[float]")
```

Reply:
I googled a bit and didn't find any typing styling like `ArrayLike[...]`. So I digged into the numpy source code. I think the element types are already specified in this case. 

https://github.com/numpy/numpy/blob/be296e2c032f7fbf30534fbe40af43c9f6b6f914/numpy/_typing/_array_like.py#L65-L93

Reply:
Will solve in #78